pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -236,6 +338,11 @@ func (d *PmemCSIDriver) getDeploymentObjects() ([]apiruntime.Object, error) {
 		d.getNodeDaemonSet(),
 	)
 
+	if operatorVersion == TestVersion {
+		// create a dummy config map
+		objects = append(objects, d.getConfigMap())
+	}
```

Comment:
No test code in the production binary, please.

If you want to test that obsolete objects are removed, then you can let the operator install the driver, shut down the operator, add some extra object *from the test code*, then restart the operator.

Reply:
> If you want to test that obsolete objects are removed, then you can let the operator install the driver, shut down the operator, add some extra object _from the test code_, then restart the operator.

You mean the operator must delete that object created by the testcode? How could the operator know that it should that object crated externally which is not owned by the deployment?

Reply:
> How could the operator know that it should that object crated externally which is not owned by the deployment?

You can create that extra object with the deployment as owner. As long as it is of a kind which the operator supports, it'll find it - after all, that is the code that we want to test.

Reply:
You seem to be under the impression that only the controller can set the ownership relationship. That is not the case. The owner reference is a field (almost) like any other, with the main RBAC check being that write permission is needed for the object where the field is being set. The additional check is that blocking deletion of other objects by referencing them also requires write permission for that other object (more specifically, its finalizers). None of that is a problem for the test code because it runs with admin rights.

Reply:
Now I rewrote the test as you suggested, it creates an object explicitly, and checks if that gets deleted on a fresh reconcile.